interface/ceed.c

```diff
@@ -15,6 +15,7 @@
 // testbed platforms, in support of the nation's exascale computing imperative.
 
 #define _POSIX_C_SOURCE 200112
+#include <ceed.h>
```

Comment:
Really, `ceed.h` is included in `ceed-backed.h`, which is included in `ceed-impl.h`, so perhaps everything in `interface/*` should just include `ceed-impl.h` and everything in `backends/*` should just include `ceed-backend.h`?

Reply:
That's the PETSc style and I think would be fine here. I moved this because feature test macros need to be defined first. Or actually, heh, my editor/clangd automatically added `ceed.h` up above the feature test macro and I moved it down without realizing it just doesn't need to be there. (A lot of style guides say you should directly include what you use, rather than relying on transitive dependencies.)

Reply:
I don't have strong feelings one way or another. I'm fine with including all used headers for clarity if that is a more typical style convention - I just object to the current inconsistency even though it has no effect on the code itself and this change happened to bring that inconsistency to my attention

I can spin up a short PR to make the includes all match in style